src/uu/head/src/head.rs

```diff
@@ -1,240 +1,557 @@
+        NUM bytes of each file\
+        ";
+    pub const LINES_NAME: &str = "LINES";
+    pub const LINES_HELP: &str = "\
+        print the first NUM lines instead of the first 10;\n\
+        with the leading '-', print all but the last\n\
+        NUM lines of each file\
+        ";
+    pub const QUIET_NAME: &str = "QUIET";
+    pub const QUIET_HELP: &str = "never print headers giving file names";
+    pub const VERBOSE_NAME: &str = "VERBOSE";
+    pub const VERBOSE_HELP: &str = "always print headers giving file names";
+    pub const ZERO_NAME: &str = "ZERO";
+    pub const ZERO_HELP: &str = "line delimiter is NUL, not newline";
+    pub const FILES_NAME: &str = "FILE";
+}
+mod parse;
+mod split;
+
+fn app<'a>() -> App<'a, 'a> {
+    App::new(executable!())
+        .version(VERSION)
+        .about(ABOUT)
+        .usage(USAGE)
+        .arg(
+            Arg::with_name(options::BYTES_NAME)
+                .short("c")
+                .long("bytes")
+                .value_name("[-]NUM")
+                .takes_value(true)
+                .help(options::BYTES_HELP)
+                .overrides_with_all(&[options::BYTES_NAME, options::LINES_NAME])
+                .allow_hyphen_values(true),
+        )
+        .arg(
+            Arg::with_name(options::LINES_NAME)
+                .short("n")
+                .long("lines")
+                .value_name("[-]NUM")
+                .takes_value(true)
+                .help(options::LINES_HELP)
+                .overrides_with_all(&[options::LINES_NAME, options::BYTES_NAME])
+                .allow_hyphen_values(true),
+        )
+        .arg(
+            Arg::with_name(options::QUIET_NAME)
+                .short("q")
+                .long("--quiet")
+                .visible_alias("silent")
+                .help(options::QUIET_HELP)
+                .overrides_with_all(&[options::VERBOSE_NAME, options::QUIET_NAME]),
+        )
+        .arg(
+            Arg::with_name(options::VERBOSE_NAME)
+                .short("v")
+                .long("verbose")
+                .help(options::VERBOSE_HELP)
+                .overrides_with_all(&[options::QUIET_NAME, options::VERBOSE_NAME]),
+        )
+        .arg(
+            Arg::with_name(options::ZERO_NAME)
+                .short("z")
+                .long("zero-terminated")
+                .help(options::ZERO_HELP)
+                .overrides_with(options::ZERO_NAME),
+        )
+        .arg(Arg::with_name(options::FILES_NAME).multiple(true))
+}
+#[derive(PartialEq, Debug, Clone, Copy)]
+pub enum Modes {
     Lines(usize),
-    NLines(usize),
+    Bytes(usize),
 }
 
-struct Settings {
-    mode: FilterMode,
-    verbose: bool,
-    zero_terminated: bool,
+pub fn parse_mode<F>(src: &str, closure: F) -> Result<(Modes, bool), String>
+where
+    F: FnOnce(usize) -> Modes,
+{
+    match parse::parse_num(src) {
+        Ok((n, last)) => Ok((closure(n), last)),
+        Err(reason) => match reason {
```

Comment:
would it be possible to add a test covering this case?

Reply:
done